stripe/src/test/java/com/stripe/android/StripeTest.java

```diff
@@ -248,6 +253,204 @@ public void createBankAccountTokenSynchronous_withValidBankAccount_returnsToken(
+        Stripe stripe = new Stripe(mContext);
+        SourceParams bitcoinParams = SourceParams.createBitcoinParams(1000L, "usd", "abc@def.com");
+        try {
+            Source bitcoinSource =
+                    stripe.createSourceSynchronous(bitcoinParams, FUNCTIONAL_SOURCE_PUBLISHABLE_KEY);
+            assertNotNull(bitcoinSource);
+            assertNotNull(bitcoinSource.getId());
+            assertNotNull(bitcoinSource.getClientSecret());
+            assertEquals(Source.BITCOIN, bitcoinSource.getType());
+            assertEquals(1000L, bitcoinSource.getAmount().longValue());
+            assertNotNull(bitcoinSource.getSourceTypeData());
+            assertNotNull(bitcoinSource.getOwner());
+            assertEquals("abc@def.com", bitcoinSource.getOwner().getEmail());
+        } catch (StripeException stripeEx) {
+            fail("Unexpected error: " + stripeEx.getLocalizedMessage());
+        }
+    }
+
+    @Test
+    public void createSourceSynchronous_withBancontactParams_passesIntegrationTest() {
+        Stripe stripe = new Stripe(mContext);
+        SourceParams bancontactParams = SourceParams.createBancontactParams(
+                1000L, "John Doe", "example://path", "a statement described");
+        try {
+            Source bancontactSource =
+                    stripe.createSourceSynchronous(bancontactParams, FUNCTIONAL_SOURCE_PUBLISHABLE_KEY);
+            assertNotNull(bancontactSource);
+            assertNotNull(bancontactSource.getId());
+            assertNotNull(bancontactSource.getClientSecret());
+            assertEquals(Source.BANCONTACT, bancontactSource.getType());
+            assertEquals(1000L, bancontactSource.getAmount().longValue());
+            assertNotNull(bancontactSource.getSourceTypeData());
+            assertNotNull(bancontactSource.getOwner());
+            assertNotNull(bancontactSource.getRedirect());
+            assertEquals("John Doe", bancontactSource.getOwner().getName());
+            assertEquals("example://path", bancontactSource.getRedirect().getReturnUrl());
+        } catch (StripeException stripeEx) {
+            fail("Unexpected error: " + stripeEx.getLocalizedMessage());
+        }
+    }
+
+    @Test
+    public void createSourceSynchronous_withCardParams_passesIntegrationTest() {
+        Stripe stripe = new Stripe(mContext);
+        Card card = new Card(CardInputTestActivity.VALID_VISA_NO_SPACES, 12, 2050, "123");
+        SourceParams params = SourceParams.createCardParams(card);
+        try {
+            Source cardSource =
+                    stripe.createSourceSynchronous(params, FUNCTIONAL_SOURCE_PUBLISHABLE_KEY);
+            assertNotNull(cardSource);
+            assertNotNull(cardSource.getClientSecret());
+            assertNotNull(cardSource.getId());
+            assertEquals(Source.CARD, cardSource.getType());
+            assertNotNull(cardSource.getSourceTypeData());
```

Comment:
it's probably a good idea to test this a bit more – you could a billing address to the card, and then assert that all the individual card info fields are making the full round trip. [0] 

[0] https://github.com/stripe/stripe-ios/blob/master/Tests/Tests/STPSourceFunctionalTest.m#L74